### PR TITLE
Generate a Gallery demo transition-durations performance histogram

### PR DESCRIPTION
**Pull request by HansMuller**
Save the duration of the first timeline Frame event that follows a 'Start Transition' event. The Gallery app adds a 'Start Transition' event when a demo is launched.

To run the benchmark app:

```
cd ./examples/flutter_gallery
flutter drive --trace-startup --profile --target=test_driver/transitions_perf.dart
```

The benchmark takes about 2 minutes to complete on a Nexus 5x.

The data is saved in `examples/flutter_gallery/build/transition_durations.timeline.json`.

Currently each gallery demo app is launched twice, both Frame durations (microseconds) are saved. Because of startup effects like dynamic compilation, we expect the second duration to be smaller than the first. We expect both durations to be less than 100ms.

### CODE REVIEW

**Comment by HansMuller** [comment]
@yjbanov

File: examples/flutter_gallery/test_driver/transitions_perf_test.dart [hunk 1]
```diff
@@ -46,6 +50,41 @@ const List<String> demoNames = const <String>[
   'Typography'
 ];
 
+
+Future<Null> saveDurationsHistogram(List<Map<String, dynamic>> events) async {
+  final Map<String, List<int>> durations = new Map<String, List<int>>();
```
Comment (yjbanov) [inline examples/flutter_gallery/test_driver/transitions_perf_test.dart]: OMG, this looks worse than Java. Not your fault. We really need to relax our style guide for obvious cases.

**Comment by yjbanov** [comment]
LGTM

File: examples/flutter_gallery/test_driver/transitions_perf_test.dart (hunk 1 above)
Comment (HansMuller) [inline examples/flutter_gallery/test_driver/transitions_perf_test.dart]: Yeah, in cases like this I wish for var permission.

Comment (eseidelGoogle) [inline examples/flutter_gallery/test_driver/transitions_perf_test.dart]: @Hixie

**Pull request merged**